packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -3016,3 +3155,266 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
+/// [SelectableRegionScope.maybeOf].
+///
+/// This value under a [SelectableRegion] is updated frequently
+/// during selection gestures such as clicks and taps to select
+/// and keyboard shortcuts.
+enum SelectableRegionSelectionStatus {
+  /// Indicates the selection under a [SelectableRegion] is changing.
+  ///
+  /// A [SelectableRegion]s selection is changing when it is being
+  /// updated by user through selection gestures and keyboard shortcuts.
+  /// For example, during a text selection drag with a click + drag,
+  /// a [SelectableRegion]s selection is considered changing until
+  /// the user releases the click, then it will be considered finalized.
+  changing,
+
+  /// Indicates the selection under a [SelectableRegion] is finalized.
+  ///
+  /// A [SelectableRegion]s selection is finalized when it is no longer
+  /// being updated by the user through selection gestures or keyboard
+  /// shortcuts. For example, the selection will be finalized on a mouse
+  /// drag end, touch long press drag end, a single click to collapse the
+  /// selection, a double click/tap to select a word, ctrl + A / cmd + A to
+  /// select all, or a triple click/tap to select a paragraph.
+  finalized,
+}
+
+/// Notifies its listeners when the [SelectableRegion] that created this object
+/// is changing or finalizes its selection.
+///
+/// To access the [_SelectableRegionSelectionStatusNotifier] from the nearest [SelectableRegion]
+/// ancestor, use [SelectableRegionScope.maybeOf].
+final class _SelectableRegionSelectionStatusNotifier extends ChangeNotifier implements ValueListenable<SelectableRegionSelectionStatus> {
+  _SelectableRegionSelectionStatusNotifier._();
+
+  SelectableRegionSelectionStatus _selectableRegionSelectionStatus = SelectableRegionSelectionStatus.finalized;
+  /// The current value of the [SelectableRegionSelectionStatus] of the [SelectableRegion]
+  /// that owns this object.
+  ///
+  /// Defaults to [SelectableRegionSelectionStatus.finalized].
+  @override
+  SelectableRegionSelectionStatus get value => _selectableRegionSelectionStatus;
+
+  /// Sets the [SelectableRegionSelectionStatus] for the [SelectableRegion] that
+  /// owns this object.
+  ///
+  /// Listeners are notified even if the value did not change.
+  @protected
+  set value(SelectableRegionSelectionStatus newStatus) {
+    _selectableRegionSelectionStatus = newStatus;
+    notifyListeners();
+  }
+}
+
+/// Notifies its listeners when the selection under a [SelectableRegion] or
+/// [SelectionArea] is being changed or finalized.
+///
+/// Use [SelectableRegionScope.maybeOf], to access the [ValueListenable] of type
+/// [SelectableRegionSelectionStatus] under a [SelectableRegion]. Its listeners
+/// will be called even when the value of the [SelectableRegionSelectionStatus]
+/// does not change.
+final class SelectableRegionScope extends InheritedNotifier<ValueListenable<SelectableRegionSelectionStatus>> {
```

Comment:
At some point we will open on SelectableRegionState , and will likely need similar things to look up the state from the subtree. Do you think that may have potential overlap here?

Reply:
I thought about this and agree that we may want to expose more state from `SelectableRegion` in the future, though at this time i'm not sure what an interface for that would look like / how much we want to expose. What if we change the name of this to `SelectableRegionSelectionStatusScope`, to indicate its specific use, and in the future when use-cases come up for exposing more state from `SelectableRegion` we can deprecate `SelectableRegionSelectionStatusScope?` for a more robust solution.

Reply:
I also updated `SelectableRegionScope` to be an `InheritedWidget` instead of `InheritedNotifier`. In my examples I was causing unnecessary rebuilds because I registered a dependency to `SelectableRegionScope` in `didChangeDependencies`, and I also added a listener to the notifier. So I would get a notification from the `InheritedNotifier`, and I would also get a notification from the notifier itself. I think by changing to `InheritedWidget` the user has more control over rebuilds, instead of rebuilding on every change to the notifier when set up in `didChangeDependencies` or `build`.